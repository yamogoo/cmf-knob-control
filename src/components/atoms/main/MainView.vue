<template lang="pug">
main(
  :class="[`${MAIN_APP_PREFIX}-main-view`]"
)
  slot
</template>

<script setup lang="ts">
import { MAIN_APP_PREFIX } from '@/config'
</script>

<style lang="scss">
$main-view: (
  sm: (),
  md: (),
  lg: ()
);

.#{$MAIN_APP_PREFIX}-main-view {
  position: relative;
  display: block;
  @include box(100%);
  background-color: none;
  overflow: hidden;
}
</style>
