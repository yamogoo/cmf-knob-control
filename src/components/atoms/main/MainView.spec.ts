import { mount } from '@vue/test-utils'
import { describe, expect, test } from 'vitest'

import MainView from './MainView.vue'

describe('MainView', () => {
  test.each(['<p>Slot Content</p>'])('should render default slot (%s)', (content) => {
    const component = mount(MainView, {
      slots: {
        default: content
      }
    })

    const slotHtml = component.html()
    expect(slotHtml).toContain(content)
    expect(slotHtml).toMatchSnapshot()
  })
})
