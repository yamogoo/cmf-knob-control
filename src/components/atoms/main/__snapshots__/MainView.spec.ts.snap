// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainView > should render default slot (<p>Slot Content</p>) 1`] = `
"<main class="mogoo-app-main-view">
  <p>Slot Content</p>
</main>"
`;
